eth/handler.go

```diff
@@ -946,8 +946,9 @@ func (h *handler) minedBroadcastLoop() {
 			if obj == nil {
 				continue
 			}
-			if ev, ok := obj.Data.(core.NewMinedBlockEvent); ok {
-				h.BroadcastBlock(ev.Block, true)  // First propagate block to peers
+			if ev, ok := obj.Data.(core.NewSealedBlockEvent); ok {
```

Comment:
I am not sure how exactly this improves things? Could you clarify bit more? 
Now we are waiting for a sealing to happen before broadcasting rather than mining to happen before broadcasting it. 
But in `(w *worker) resultLoop()` function, there are several checks made before a mining event is emitted which could fail e.g. double sign. Wouldn't that be a problem?